### DISCUSSION TITLE
use ShowWindow with SW_HIDE

### DISCUSSION THREAD

**Discussion by kahotv** [ocornut/imgui#8257] [New Users Build/Link/Run issues ONLY!]
In the example_win32_directx11 project, change `::ShowWindow(hwnd, SW_SHOWDEFAULT);` to `::ShowWindow(hwnd, SW_HIDE);` and delete the automatically generated imgui.ini file. After compiling and running the program, the ImGui window will not appear.

By comparing imgui.ini, it was found that manually adding ViewportPos and ViewportId can solve this issue. So, how can ViewportPos and ViewportId be automatically added when using SW_HIDE? Especially when running for the first time, at which point there is no imgui.ini file.

**Comment by runbozhuang** [comment]
Hi,

I believe the setting(the information of imgui.ini storing)  is generated by **InitOrLoadWindowSettings(window, settings);**
Navigate it in "static ImGuiWindow* CreateNewWindow" method in imgui.cpp

The pos and id is generated when Imgui window is created, it should not related to showWindow parameter if I am not mistaken 
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-showwindow
